lib/prism.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const Prism = require('prismjs');
+const prismLoadLanguages = require('prismjs/components/');
+
+// https://github.com/PrismJS/prism/issues/2145
+const prismComponents = require('prismjs/components');
+const prismSupportedLanguages = Object.keys(prismComponents.languages);
+
+const escapeHTML = require('./escape_html');
+const unescapeHTML = require('./unescape_html');
+
+/**
+ * Wrapper of Prism.highlight()
+ * @param {String} code
+ * @param {String} language
+ */
+function prismHighlight(code, language) {
+  console.log(Object.keys(Prism.languages));
+  // Prism has not load the language pattern
+  if (!Prism.languages[language] && prismSupportedLanguages.includes(language)) prismLoadLanguages(language);
+
+  if (Prism.languages[language]) {
+    // Prism escapes output by default
+    return Prism.highlight(unescapeHTML(code), Prism.languages[language], language);
+  }
+
+  // Current language is not supported by Prism, return origin code;
+  return escapeHTML(code);
+}
+
+/**
+ * Generate line number required HTML snippet
+ * @param {String} code - Highlighted code
+ */
+function lineNumberUtil(code) {
```

Comment:
@SukkaW Do you intend to re-implement the "line-numbers" plugin of Prism?
If yes, some functions seem to lack (such as a SPAN element with the class name "line-numbers-sizer").

Could you taste a work-around to use plugins in Node.js ?
https://github.com/PrismJS/prism/issues/1171#issuecomment-470929808

Reply:
https://github.com/PrismJS/prism/issues/1171#issuecomment-381774025

> Line Numbers: Does not work and returns early. Requires DOM.

And the current approach should be compatible with `prism-line-numbers.css`:

https://github.com/PrismJS/prism/blob/master/plugins/line-numbers/prism-line-numbers.css

Reply:
https://prismjs.com/plugins/line-numbers/

![image](https://user-images.githubusercontent.com/40715044/71494785-c5f23380-2884-11ea-83a4-1142e6e2da37.png)

As you can see, this is the DOM structure after enabling the prism's line numbers plugin. No `.line-numbers-sizer`.

Reply:
![image](https://user-images.githubusercontent.com/40715044/71518772-5feec600-28ef-11ea-9aa4-cf18c78050c5.png)

https://github.com/PrismJS/prism/blob/e20a2b711a0cd4ee7e6b49b07f7ce165df2a7980/plugins/line-numbers/prism-line-numbers.js#L48-L49

It seems that the `lineNumberSizer` will be removed.

Reply:
I agreed.